src/index.rs

```diff
@@ -1609,8 +1609,18 @@ impl Index {
     /// Returns all enum variants of the given variable.
     pub fn get_enum_variants_by_variable(&self, variable: &VariableIndexEntry) -> Vec<&VariableIndexEntry> {
         let Some(var) = self.type_index.find_type(&variable.data_type_name) else { return vec![] };
-        let DataTypeInformation::Enum { variants, .. } = var.get_type_information() else { return vec![] };
-        variants.iter().collect()
+
+        match var.get_type_information() {
+            DataTypeInformation::Enum { variants, .. } => variants.iter().collect(),
+            DataTypeInformation::Pointer { name, inner_type_name, .. } => {
```

Comment:
I would also check here that the pointer is an auto deref pointer, otherwise we start returing type variants for pointer types.

```suggestion
            DataTypeInformation::Pointer { name, inner_type_name, auto_deref: Some(_).. } => {
```

Reply:
Where would you recommend adding the unit tests? I haven't looked very deep, but I haven't figured out how properly use the `.snap` based tests.